code/modules/mob/living/basic/lavaland/mook/mook.dm

```diff
@@ -0,0 +1,405 @@
+	ai_controller = /datum/ai_controller/basic_controller/mook
+	is_healer = FALSE
+
+/mob/living/basic/mining/mook/worker/Initialize(mapload)
+	. = ..()
+	neutral_stance = mutable_appearance(icon, "mook_axe_overlay")
+	attack_stance = mutable_appearance(icon, "axe_strike_overlay")
+	update_appearance()
+	var/datum/action/cooldown/mob_cooldown/mook_ability/mook_leap/leap = new(src)
+	leap.Grant(src)
+	ai_controller.set_blackboard_key(BB_MOOK_LEAP_ABILITY, leap)
+
+/mob/living/basic/mining/mook/worker/attack_sequence(atom/target)
+	. = ..()
+	if(. & COMPONENT_HOSTILE_NO_ATTACK)
+		return
+
+	if(attack_state == MOOK_ATTACK_STRIKE)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	change_combatant_state(state = MOOK_ATTACK_STRIKE)
+	addtimer(CALLBACK(src, PROC_REF(change_combatant_state), MOOK_ATTACK_NEUTRAL), 0.3 SECONDS)
+
+/mob/living/basic/mining/mook/worker/update_overlays()
+	. = ..()
+	if(stat == DEAD)
+		return
+
+	switch(attack_state)
+		if(MOOK_ATTACK_STRIKE)
+			. += attack_stance
+		if(MOOK_ATTACK_NEUTRAL)
+			. += neutral_stance
+
+/mob/living/basic/mining/mook/worker/bard
+	desc = "It's holding a guitar?"
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	gender = MALE
+	attack_sound = 'sound/weapons/stringsmash.ogg'
+	death_sound = 'sound/voice/mook_death.ogg'
+	ai_controller = /datum/ai_controller/basic_controller/mook/bard
+	///our guitar
+	var/obj/item/instrument/guitar/held_guitar
+
+/mob/living/basic/mining/mook/worker/bard/Initialize(mapload)
+	. = ..()
+	neutral_stance = mutable_appearance(icon, "bard_overlay")
+	attack_stance = mutable_appearance(icon, "bard_strike")
+	held_guitar = new(src)
+	ai_controller.set_blackboard_key(BB_SONG_INSTRUMENT, held_guitar)
+	update_appearance()
+
+/mob/living/basic/mining/mook/worker/tribal_chief
+	name = "tribal chief"
+	desc = "Acknowledge him!"
+	gender = MALE
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+	ai_controller = /datum/ai_controller/basic_controller/mook/tribal_chief
+	///overlay in our neutral state
+	var/mutable_appearance/chief_neutral
+	///overlay in our striking state
+	var/mutable_appearance/chief_strike
+	///overlay in our active state
+	var/mutable_appearance/chief_active
+	///overlay in our warmup state
+	var/mutable_appearance/chief_warmup
+
+/mob/living/basic/mining/mook/worker/tribal_chief/Initialize(mapload)
+	. = ..()
+	chief_neutral = mutable_appearance(icon, "mook_chief")
+	chief_strike = mutable_appearance(icon, "mook_chief_strike")
+	chief_active = mutable_appearance(icon, "mook_chief_leap")
+	chief_warmup = mutable_appearance(icon, "mook_chief_warmup")
+	update_appearance()
+
+/mob/living/basic/mining/mook/worker/tribal_chief/update_overlays()
+	. = ..()
+	if(stat == DEAD)
+		return
+	switch(attack_state)
+		if(MOOK_ATTACK_NEUTRAL)
+			. += chief_neutral
+		if(MOOK_ATTACK_WARMUP)
+			. += chief_warmup
+		if(MOOK_ATTACK_ACTIVE)
+			. += chief_active
+		if(MOOK_ATTACK_STRIKE)
+			. += chief_strike
+
+/datum/action/cooldown/mob_cooldown/mook_ability
```

Comment:
i think you should make a new file like `mook_abilities.dm` this file is too long. i see you're reliant on the defines you set in this file but if you either

A) make them global defines (preferred)
B) rename this file to `_mook.dm` and just remove the `#undef`s at the bottom

then the linter won't get mad at you

Reply:
do u know where i can put these defines? i been looking for a home for them since i started this project

Reply:
i just smack it somewhere in the mobs define file or make a new one in the folder